include/ui.js

```diff
@@ -810,7 +812,17 @@ var UI;
             var host = $D('noVNC_host').value;
             var port = $D('noVNC_port').value;
             var password = $D('noVNC_password').value;
+            var token = $D('noVNC_token').value;
             var path = $D('noVNC_path').value;
+
+            //if token is in path then ignore the new token variable
+            if (token && !(path.indexOf("token") > -1)){
```

Comment:
Doesn't this cause the token insertion not to trigger if I have
`path = "/path"`, in which case a path of `path?path=[foo]` is perfectly valid.